### PR TITLE
test_fd

### PR DESCRIPTION
**Pull request by mloubout**
New test for the symbolic finite differences against polynomial.
Allowed me to find a sign error.

The `sign` is removed from `finite_differences` as it was a hack for adjoints. The time-axis PR will introduce proper adjoint flag.

IN PROGRESS

### CODE REVIEW

**Comment by mloubout** [comment]
This needs PR 176 first

File: examples/tti/tti_operators.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def ForwardOperator(model, u, v, src, rec, damp, data, time_order=2,
         kwargs.pop('dle', None)
 
         op = Operator(nt, m.shape, stencils=stencils, subs=[subs, subs],
-                      spc_border=spc_order/2, time_order=time_order,
+                      spc_border=spc_order, time_order=time_order,
```
Comment (FabioLuporini) [inline examples/tti/tti_operators.py]: ugh?

File: tests/test_fd.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    # dummy axis dimension
+    nx = 100
+    xx = np.linspace(-1, 1, nx)
+    dx = xx[1] - xx[0]
+    # Symbolic data
+    u = DenseData(name="u", shape=(nx,), space_order=space_order, dtype=np.float32)
+    du = DenseData(name="du", shape=(nx,), space_order=space_order, dtype=np.float32)
+    # Define polynomial with exact fd
+    y = symbols('y')
+    h = symbols('h')
```
Comment (FabioLuporini) [inline tests/test_fd.py]: ``y, h = symbols('y h')``
 just to be pedantic

File: tests/test_fd.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from sympy import Eq, diff, symbols
+
+from devito import StencilKernel, clear_cache
+from devito.interfaces import DenseData
+
+
+@pytest.mark.parametrize('space_order', [2, 4, 6, 8, 10, 12, 14, 16, 18, 20])
+# Only test x and t as y and z are the same as x
+@pytest.mark.parametrize('derivative', ['dx', 'dxl', 'dxr', 'dx2'])
+def test_fd_space(derivative, space_order):
```
Comment (FabioLuporini) [inline tests/test_fd.py]: can you add a docstring explaining precisely what this test does

Comment (FabioLuporini) [inline tests/test_fd.py]: also, use a more sensible file name?

File: examples/tti/tti_operators.py (hunk 1 above)
Comment (mloubout) [inline examples/tti/tti_operators.py]: Non-symmetric stencil in TTI, I don't know when the /2 came back but it will segfault with it

**Comment by mloubout** [comment]
This is good to go?

**Comment by FabioLuporini** [comment]
It's a yes for me. You'll have to rebase on master first though. Is this OK with you @mlange05 ?

**Comment by mlange05** [comment]
Yes, looks good to me. Merging....

**Pull request merged**